### PR TITLE
Addition of new auth API (preview) and ApiInformation interface.

### PR DESCRIPTION
**Pull request by sumurthy**
Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition: N/A
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.microsoft.com/en-us/office/dev/add-ins/?view=excel-js-preview
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. N/A
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed. N/A

If removing a declaration: N/A
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/office-runtime/index.d.ts [hunk 1]
```diff
@@ -13,101 +13,191 @@ Copyright (c) Microsoft Corporation
...
+   */
+  const auth: Auth;
+  /**
+   * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the getAccessTokenAsync method.
+   */
+  interface AuthOptions {
+    /**
+     * Allows Office to get an access token silectly or through interactive consent, if one is required
+     * If set to false, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error
+     * If set to true, Office will show an interactive consent UI after it fails to silently get an access token
```
Comment (Rick-Kirkham) [inline types/office-runtime/index.d.ts]: We need to add that the prompt will only allow consent to the AAD profile scope, not to any Microsoft Graph scopes.

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]

File: types/office-runtime/office-runtime-tests.ts [hunk 2]
```diff
@@ -7,19 +7,24 @@ OfficeRuntime.storage.getItem("foo");
...
     onClose: () => {
         console.log("closed");
     }
- });
+});
+
+// Check for the support of the requirement-set (note that it only accepts strings.)
+const supported = OfficeRuntime.apiInformation.isSetSupported("ExcelApi", "1.9");
+
+// For SSO auth API, refer to the public documentation at the site: https://docs.microsoft.com/en-us/office/dev/add-ins/?view=excel-js-preview
```
Comment (AlexJerabek) [inline types/office-runtime/office-runtime-tests.ts]: Please remove the "/en-us" from the URL.

File: types/office-runtime/index.d.ts [hunk 3]
```diff
@@ -3,111 +3,194 @@
...
+  /** Object representing the dialog box. */
+  interface Dialog {
+    /** Method to close a dialog box. Returns a Promise. */
+    close(): Promise<void>;
+  }
+  /** Provides display options and actions a dialog box may take. */
+  interface DisplayWebDialogOptions {
+    /**
+     * Optional parameter that determines whether the dialog box displays as a popup (false) or within an IFrame (true).
+     * This setting is only applicable to custom functions running on Excel Online.
```
Comment (davidchesnut) [inline types/office-runtime/index.d.ts]:
```suggestion
     * This setting is only applicable to custom functions running on Excel on the web.
```

File: types/office-runtime/index.d.ts [hunk 4]
```diff
@@ -3,111 +3,194 @@
...
+     */
+    onRuntimeError?: (error: Error, dialog?: Dialog) => void;
+  }
+  /**
+   * Contains authorization related APIs.
+   * @beta
+   */
+  const auth: Auth;
+  /**
+   * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the getAccessToken method.
```
Comment (davidchesnut) [inline types/office-runtime/index.d.ts]:
```suggestion
   * Provides options for the user experience when Office obtains an access token to the add-in from Microsoft Azure Active Directory (Azure AD) v. 2.0 with the getAccessToken method.
```

File: types/office-runtime/index.d.ts [hunk 5]
```diff
@@ -3,111 +3,194 @@
...
+   * @beta
+   */
+  interface AuthOptions {
+    /**
+     * Allows Office to get an access token silectly or through interactive consent, if one is required.
+     * If set to false, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
+     * If set to true, Office will show an interactive consent UI after it fails to silently get an access token.
+     * The prompt will only allow consent to the AAD profile scope, not to any Microsoft Graph scopes.
```
Comment (davidchesnut) [inline types/office-runtime/index.d.ts]:
```suggestion
     * The prompt will only allow consent to the Microsoft Azure Active Directory (Azure AD) profile scope, not to any Microsoft Graph scopes.
```

File: types/office-runtime/index.d.ts [hunk 6]
```diff
@@ -3,111 +3,194 @@
...
+     * call of getAccessToken after Microsoft Graph has sent an error requesting the additional factor and containing the string that should
+     * be used with the authChallenge option.
+     */
+    authChallenge?: string;
+    /**
+     * A user-defined item of any type that is returned, unchanged, in the asyncContext property of the AsyncResult object that is passed to a callback.
+     */
+    asyncContext?: any;
+    /**
+     * Causes Office to return descriptive error when the add-in wants to access MS Graph and the user/admin has not granted consent to MS Graph scopes.
```
Comment (davidchesnut) [inline types/office-runtime/index.d.ts]:
```suggestion
     * Causes Office to return descriptive error when the add-in wants to access MS Graph and the user/admin has not granted consent to Microsoft Graph scopes.
```

File: types/office-runtime/index.d.ts [hunk 7]
```diff
@@ -3,111 +3,194 @@
...
+     */
+    forMSGraphAccess?: boolean;
+  }
+  /**
+   * Interface that contains authorization related APIs.
+   * @beta
+   */
+  interface Auth {
+    /**
+     * Calls the Azure Active Directory V 2.0 endpoint to get an access token to your add-in's web application. Enables add-ins to identify users.
```
Comment (davidchesnut) [inline types/office-runtime/index.d.ts]:
```suggestion
     * Calls the Microsoft Azure Active Directory (Azure AD) V 2.0 endpoint to get an access token to your add-in's web application. Enables add-ins to identify users.
```

**Review by davidchesnut — APPROVED** [review]
I put in a few acronym changes.

**Review by Rick-Kirkham — APPROVED** [review]

**Review by AlexJerabek — APPROVED** [review]

**Comment by sumurthy** [comment]
Please merge. 
I don't know how to deal with the TypeScript version not found error in the benchmark automation tests.

**Comment by AlexJerabek** [comment]
@sandersn
Could you please investigate these benchmark failures? They appear to be unrelated to the changes in @sumurthy's PR.

**Comment by sandersn** [comment]
Yep, they are not related, but a break of the benchmark itself when the Typescript 3.7 RC released.

**Pull request merged**